### PR TITLE
fix: Preserves sort as a query parameter

### PR DESCRIPTION
**Pull request by wmak**
- This change was introduced so that when a user navigates from the
  Issues Stream -> Issues Details -> Issues Stream the sort order of
  Issues would be preserved.

### CODE REVIEW

**Review by billyvg — CHANGES_REQUESTED** [review]
So what I think we need to do is, is the issues list, these links need to append `sort` key if it exists on the url parameters.

image

Then in the follow issue details screen, if `sort` exists in url params, when we click this button, it should preserve `sort`:

image

**Comment by wmak** [comment]
Hm @billyvg that is happening by adding `sort` to `URL_PARAM`, since we will then keep sort as part of the `globalQuery` when `extractSelectionParameters` is called by `GlobalSelectionLink`.

Is it that `sort` shouldn't be a part of that `globalQuery` and we want to treat it differently than the other url parameters?

**Comment by billyvg** [comment]
@wmak correct, we don't necessarily want to treat it differently, it's just that there is no "sort" functionality in Global Selection Header so it may confuse people in the future when they find that constant in there.

File: src/sentry/static/sentry/app/components/eventOrGroupHeader.jsx [hunk 1]
```diff
@@ -44,13 +44,14 @@ class EventOrGroupHeader extends React.Component {
...
 
     if (includeLink) {
-      const locationWithProject = {...this.props.location};
-      const query =
-        locationWithProject.query.project !== undefined
-          ? {
-              query: this.props.query,
-            }
-          : {query: this.props.query, _allp: 1}; //This appends _allp to the URL parameters if they have no project selected ("all" projects included in results). This is so that when we enter the issue details page and lock them to a project, we can properly take them back to the issue list page with no project selected (and not the locked project selected)
+      const locationWithQueries = {...this.props.location};
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/eventOrGroupHeader.jsx]: Why do you need a clone of the location? You don't mutate the copy.

Comment (billyvg) [reply src/sentry/static/sentry/app/components/eventOrGroupHeader.jsx]: This is old code, but yeah does not look like we need to clone this.

**Review by markstory — APPROVED** [review]

**Review by billyvg — APPROVED** [review]
👍 Thanks for fixing this!

**Pull request merged**